pandas/tests/indexes/test_multi.py

```diff
@@ -154,7 +154,7 @@ def assert_matching(actual, expected):
             # as much as possible
             self.assertEqual(len(actual), len(expected))
             for act, exp in zip(actual, expected):
-                act = np.asarray(act)
+                act = np.asarray(act, dtype=np.object_)
```

Comment:
But when I take out the coercion to `np.object_`, the test `test_set_levels` fails with 
```
E       AssertionError: numpy array are different
E       
E       Attribute "dtype" are different
E       [left]:  object
E       [right]: |S4
```
This is in the test
```
        # level changing [w/o mutation]
        ind2 = self.index.set_levels(new_levels)
        assert_matching(ind2.levels, new_levels)
        assert_matching(self.index.levels, levels)
```